applications/StructuralMechanicsApplication/custom_processes/assign_nodal_elements_to_nodes_process.cpp

```diff
@@ -0,0 +1,303 @@
+        if (!mThisParameters["nodal_rotational_stiffness"][0].IsNull()) {
+            nodal_rotational_stiffness[0] = mThisParameters["nodal_rotational_stiffness"][0].GetDouble();
+        }
+        if (!mThisParameters["nodal_rotational_stiffness"][1].IsNull()) {
+            nodal_rotational_stiffness[1] = mThisParameters["nodal_rotational_stiffness"][1].GetDouble();
+        }
+        if (!mThisParameters["nodal_rotational_stiffness"][2].IsNull()) {
+            nodal_rotational_stiffness[2] = mThisParameters["nodal_rotational_stiffness"][2].GetDouble();
+        }
+
+        p_properties->SetValue(NODAL_ROTATIONAL_STIFFNESS, nodal_rotational_stiffness);
+    }
+    array_1d<double, 3> nodal_damping_ratio = ZeroVector(3);
+    if (!mThisParameters["nodal_damping_ratio"][0].IsNull() || !mThisParameters["nodal_damping_ratio"][1].IsNull() || !mThisParameters["nodal_damping_ratio"][2].IsNull()) {
+        if (!mThisParameters["nodal_damping_ratio"][0].IsNull()) {
+            nodal_damping_ratio[0] = mThisParameters["nodal_damping_ratio"][0].GetDouble();
+        }
+        if (!mThisParameters["nodal_damping_ratio"][1].IsNull()) {
+            nodal_damping_ratio[1] = mThisParameters["nodal_damping_ratio"][1].GetDouble();
+        }
+        if (!mThisParameters["nodal_damping_ratio"][2].IsNull()) {
+            nodal_damping_ratio[2] = mThisParameters["nodal_damping_ratio"][2].GetDouble();
+        }
+
+        p_properties->SetValue(NODAL_DAMPING_RATIO, nodal_damping_ratio);
+    }
+    array_1d<double, 3> nodal_rotational_damping_ratio = ZeroVector(3);
+    if (!mThisParameters["nodal_rotational_damping_ratio"][0].IsNull() || !mThisParameters["nodal_rotational_damping_ratio"][1].IsNull() || !mThisParameters["nodal_rotational_damping_ratio"][2].IsNull()) {
+        if (!mThisParameters["nodal_rotational_damping_ratio"][0].IsNull()) {
+            nodal_rotational_damping_ratio[0] = mThisParameters["nodal_rotational_damping_ratio"][0].GetDouble();
+        }
+        if (!mThisParameters["nodal_rotational_damping_ratio"][1].IsNull()) {
+            nodal_rotational_damping_ratio[1] = mThisParameters["nodal_rotational_damping_ratio"][1].GetDouble();
+        }
+        if (!mThisParameters["nodal_rotational_damping_ratio"][2].IsNull()) {
+            nodal_rotational_damping_ratio[2] = mThisParameters["nodal_rotational_damping_ratio"][2].GetDouble();
+        }
+
+        p_properties->SetValue(NODAL_ROTATIONAL_DAMPING_RATIO, nodal_rotational_damping_ratio);
+    }
+
+    // The number of elements
+    const SizeType number_elements = r_root_model_part.NumberOfElements();
+    const auto it_elem_begin = r_root_model_part.ElementsBegin();
+
+    // Reorder ids
+    IndexPartition<std::size_t>(number_elements).for_each([&it_elem_begin](std::size_t i) {
+        auto it_elem = it_elem_begin + i;
+        it_elem->SetId(i + 1);
+    });
```

Comment:
As far as I understand, in here you change the already existent element ids to be consecutive. On top of the fact that, as it is right now, I don't think this works in MPI, I don't really see the need for doing so.